src/library/scala/collection/immutable/Stream.scala

```diff
@@ -471,7 +471,7 @@ object Stream extends SeqFactory[Stream] {
 
   def empty[A]: Stream[A] = Empty
 
-  override def newBuilder[A]: mutable.Builder[A, Stream[A]] = ArrayBuffer.newBuilder[A].mapResult(array => from(array))
+  override def newBuilder[A]: mutable.Builder[A, Stream[A]] = Iterator.newBuilder[A].mapResult(fromIterator)
```

Comment:
this could use LazyList couldn't it? That would align Stream and LazyList more, which I think would be good.

Reply:
no reason to. `Stream` does its own memoizing, so it's okay that the built iterator can only be used once. `View`, however, needs a backing collection that will memoize the elements.

Reply:
and `LazyList`'s builder does not allow you to call `result()` multiple times in a row, so switching to it would not fix the failing test